docs/specification/README.md

```diff
@@ -163,11 +163,17 @@ The message encoding and schema must match that of the Channel record correspond
 | Bytes | Name | Type | Description |
 | --- | --- | --- | --- |
 | 2 | channel_id | uint16 | Channel ID |
-| 4 | sequence | uint32 | Optional message counter assigned by publisher. If not assigned by publisher, must be recorded by the recorder. |
+| 4 | sequence | uint32 | Optional message sequence count, which can be used by readers to detect dropped or out-of-order messages. If not used, all `sequence` values in Message records with the same `channel_id` should be 0. |
 | 8 | log_time | Timestamp | Time at which the message was recorded. |
 | 8 | publish_time | Timestamp | Time at which the message was published. If not available, must be set to the log time. |
 | N | data | Bytes | Message data, to be decoded according to the schema of the channel. |
 
+!!! note "Using Message Sequence Counts"
+
+    Message sequence counts are specific to a channel, meaning that each successive Message record in an MCAP with the same channel ID should increase in `sequence` by 1. After this sequence count reaches (2<sup>32</sup> - 1), it is expected to roll over to 0.
```

Comment:
This seems overly restrictive. If I'm using topic filtering for recording so only a subset of messages on a topic are recorded into an MCAP, the `sequence` would not increment by one if I'm using sequence numbers assigned by my pub/sub middleware. I might have a policy of only recording every tenth message, so I would expect the sequence numbers to go 10, 20, 30, ...

Reply:
Those message sequence numbers may be required to make a system function. For example, sequence numbers could be associated with H.264 frame numbers in a side channel topic so `10, 20, 30, ...` values must be present for replay to work.